File: Python/13_Roman_to_Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        result = 0
        i = 0
        while i < len(s):
            if i + 1 < len(s) and s[i] == 'I' and s[i + 1] == 'V':
                result += 4
                i += 2
            elif i + 1 < len(s) and s[i] == 'I' and s[i + 1] == 'X':
                result += 9
                i += 2
            elif i + 1 < len(s) and s[i] == 'X' and s[i + 1] == 'L':
                result += 40
                i += 2
            elif i + 1 < len(s) and s[i] == 'X' and s[i + 1] == 'C':
                result += 90
                i += 2
            elif i + 1 < len(s) and s[i] == 'C' and s[i + 1] == 'D':
                result += 400
                i += 2
            elif i + 1 < len(s) and s[i] == 'C' and s[i + 1] == 'M':
                result += 900
                i += 2
            else:
                if s[i] == 'I':
                    result += 1
                elif s[i] == 'V':
                    result += 5
                elif s[i] == 'X':
                    result += 10
                elif s[i] == 'L':
                    result += 50
                elif s[i] == 'C':
                    result += 100
                elif s[i] == 'D':
                    result += 500
                elif s[i] == 'M':
                    result += 1000
                i += 1
        return result

File: C/13_Roman_to_Integer.c
int romanToInt(char* s) {
    int result = 0;
    int i;
    for(i = 0; i < strlen(s) - 1; i++)
    {
        if(s[i] == 'I' && s[i + 1] == 'V')
        {
            result = result + 4;
            i++;
        }
        else if(s[i] == 'I' && s[i + 1] == 'X')
        {
            result = result + 9;
            i++;
        }
        else if(s[i] == 'X' && s[i + 1] == 'L')
        {
            result = result + 40;
            i++;
        }
        else if(s[i] == 'X' && s[i + 1] == 'C')
        {
            result = result + 90;
            i++;
        }
        else if(s[i] == 'C' && s[i + 1] == 'D')
        {
            result = result + 400;
            i++;
        }
        else if(s[i] == 'C' && s[i + 1] == 'M')
        {
            result = result + 900;
            i++;
        }
        else if(s[i] == 'I')
            result = result + 1;
        else if(s[i] == 'V')
            result = result + 5;
        else if(s[i] == 'X')
            result = result + 10;
        else if(s[i] == 'L')
            result = result + 50;
        else if(s[i] == 'C')
            result = result + 100;
        else if(s[i] == 'D')
            result = result + 500;
        else if(s[i] == 'M')
            result = result + 1000;
    }
    if(s[i] == 'I')
        result = result + 1;
    else if(s[i] == 'V')
        result = result + 5;
    else if(s[i] == 'X')
        result = result + 10;
    else if(s[i] == 'L')
        result = result + 50;
    else if(s[i] == 'C')
        result = result + 100;
    else if(s[i] == 'D')
        result = result + 500;
    else if(s[i] == 'M')
        result = result + 1000;
    return result;
}